js/common/interrupt_error.go

```diff
@@ -0,0 +1,44 @@
+/*
+ *
+ * k6 - a next-generation load testing tool
+ * Copyright (C) 2021 Load Impact
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package common
+
+import "errors"
+
+// InterruptError is an error that halts engine execution
+type InterruptError struct {
+	Reason string
+}
+
+func (i InterruptError) Error() string {
+	return i.Reason
+}
+
```

Comment:
This will probably be better off as a pointer receiver because otherwise you somewhere might return non pointer to it and it will *still* count as error but will not pass the `IsInterruptError` below. While in order to "break" it with pointer receiver you will need to return pointer to a pointer to InterruptError which ... seems like a bad idea in the first place 

[playground link trying to showcase the problem](https://play.golang.org/p/JUgsHN_hpdp)

Reply:
Good point, thanks for explaining. Will fix.

Reply:
Resolved in recent push.